pajbot/modules/chat_alerts/raidalert.py

```diff
@@ -0,0 +1,118 @@
+import logging
+
+from pajbot.managers.handler import HandlerManager
+from pajbot.modules import BaseModule
+from pajbot.modules import ModuleSetting
+from pajbot.modules.chat_alerts import ChatAlertModule
+
+log = logging.getLogger(__name__)
+
+
+class RaidAlertModule(BaseModule):
+    ID = __name__.split(".")[-1]
+    NAME = "Raid Alert"
+    DESCRIPTION = "Prints a message in chat/whispers when a user raids your channel"
+    CATEGORY = "Feature"
+    ENABLED_DEFAULT = False
+    PARENT_MODULE = ChatAlertModule
+    SETTINGS = [
+        ModuleSetting(
+            key="chat_message",
+            label="Enable a chat message for someone who raids",
+            type="boolean",
+            required=True,
+            default=True,
+        ),
+        ModuleSetting(
+            key="new_raid",
+            label="Raid chat message | Available arguments: {username}, {num_viewers}",
+            type="text",
+            required=True,
+            placeholder="{username} just raided the channel with {num_viewers} viewers PogChamp",
+            default="{username} just raided the channel with {num_viewers} viewers PogChamp",
+            constraints={"min_str_len": 10, "max_str_len": 400},
+        ),
+        ModuleSetting(
+            key="whisper_message",
+            label="Enable a whisper message for someone who raids",
+            type="boolean",
+            required=True,
+            default=False,
+        ),
+        ModuleSetting(
+            key="whisper_after",
+            label="Whisper the message after X seconds",
+            type="number",
+            required=True,
+            placeholder="",
+            default=5,
+            constraints={"min_value": 1, "max_value": 120},
+        ),
+        ModuleSetting(
+            key="raid_whisper",
+            label="Whisper message for raids | Available arguments: {username}, {num_viewers}",
+            type="text",
+            required=True,
+            placeholder="Thank you for raiding with {num_viewers} viewers {username} <3",
+            default="Thank you for raiding with {num_viewers} viewers {username} <3",
+            constraints={"min_str_len": 10, "max_str_len": 400},
+        ),
+        ModuleSetting(
+            key="grant_points_on_raid",
+            label="Give points to user when they raid. 0 = off",
+            type="number",
+            required=True,
+            placeholder="",
+            default=0,
+            constraints={"min_value": 0, "max_value": 50000},
+        ),
```

Comment:
could we get a setting for a minimum raid size? so we can exclude 1 Viewer Andy's